Replace progress bar with updated styles in Deployments page

pierlo-upitup opened the pull request:
This fixes https://github.com/mesosphere/marathon/issues/2715

Here's a before/after gif

![progress-bar](https://cloud.githubusercontent.com/assets/1078545/11506506/0f08ebc8-9850-11e5-9ac9-4b9485efc2ef.gif)

aldipower commented:
:+1:

philipnrmn commented on src/css/components/loading-bar.less:
```diff
@@ -0,0 +1,27 @@
+  background-size: 12px 12px;
+  box-shadow: inset 0 10px 0 rgba(255, 255, 255, 0.2);
+  animation: move 2s linear infinite;
+  -ms-animation-name: move;
```
I take it that this isn't covered correctly by autoprefixing?

pierlo-upitup replied:
Good call. I simply ported this from the existing styles, but it might be a good idea. Should I try that?

philipnrmn replied:
Removing it stops it appearing in the built css file, so we should leave it in.

pierlo-upitup replied:
Thanks for testing that... but then we have a problem with the autoprefixer, I guess?

pierlo-upitup replied:
Let me rephrase: maybe it makes sense to add a comment here so we don't ask ourselves the same question later on? :tongue:

philipnrmn commented on src/css/components/loading-bar.less:
```diff
@@ -0,0 +1,27 @@
+}
+
+@keyframes move {
+  to { background-position: 12px 12px; }
```
:+1:

philipnrmn commented:
This PR might clash with #451, as it introduces old style tooltips:

![sit still, darn it](https://s3.amazonaws.com/f.cl.ly/items/1u2g1a0J033f1l1j2L1N/Screen%20Recording%202015-12-02%20at%2004.01%20pm.gif?v=c2777d21)

If both are merged we'll have two different styles on nearby pages. 

@leemunroe how should the tooltips look?

pierlo-upitup commented:
@philipnrmn really good point. I actually added the tooltips (they were not there in the Deployments page) so we might consider removing them as well.

philipnrmn commented:
The deploying bar here is rectangular, whereas our health bar has rounded corners.

pierlo-upitup commented:
Thanks for the heads up @philipnrmn ... I totally missed the border radius!

Fixed in the last commit: 

![depl](https://cloud.githubusercontent.com/assets/1078545/11536499/02b28fea-9919-11e5-81c6-89aa5c4fcfe7.gif)

Also rebased onto master so we shouldn't have any issues with the newer npm version.

leemunroe commented:
Looks great. Provides good feedback that stuff is happening.

@philipnrmn Are the tooltips different? As in the style or the content that is in them? "Deploying" LGTM.

philipnrmn commented:
@leemunroe OK, I guess if you're happy then we can go ahead. 

To me it feels weird to have gone from this:
![](https://cloud.githubusercontent.com/assets/2989362/11446432/18dc56d8-9537-11e5-9961-8b96b0dcf5b5.png)

To this:
![](https://cloud.githubusercontent.com/assets/2989362/11446433/1e19978c-9537-11e5-921f-f8be3737b40d.png)

And then introduced this:
![screen shot 2015-12-03 at 09 57 49](https://cloud.githubusercontent.com/assets/2989362/11556391/88a5681a-99a6-11e5-84b8-98d211531810.png)

leemunroe commented:
I think I misunderstood this interaction. Agreed it is confusing if there is a tooltip with "Deploying".

What we want to achieve here is to provide UI feedback to the user that we received their click and we're doing something. In this case, if they click "Rollback", we're going to stop the current deployment and about to start a new deployment. Is that accurate?

If so, no tooltip at all might make more sense for now. And we might even want to look at this being something other than a bar. But that's outside the scope of this atm.

pierlo-upitup commented:
Cool, removing the tooltip!

philipnrmn commented:
:+1: :ship:
